### DISCUSSION TITLE
Hexadecimal Unicode input in altchar

### DISCUSSION THREAD

**Discussion by bitoj** [cajhin/capsicain#22] [Ideas]
I think it would be useful to be able to enter hexadecimal character codes in the `altchar` function, as described in the Wikipedia article [here](https://en.wikipedia.org/wiki/Unicode_input#Hexadecimal_input).

Advantages:
- It will be possible to enter characters outside the current locale codepage in all Windows applications that do not actively prevent it, in contrast to decimal entry of code points above 255, which requires active support from an application.
- The notation is more in line with the Unicode standard and published character lists.

Disadvantage: it only works for users that have applied the following registry modification.

```
Windows Registry Editor Version 5.00

[HKEY_CURRENT_USER\Control Panel\Input Method]
"EnableHexNumpad"="1"
```

When this is allowed, we can make definitions like
```
COMBO G [... T... ...^ ...^] > altchar(+11D) # ĝ
```
which can be copied from the Windows character map application or Unicode publications, in addition to
```
COMBO G [... T... ...^ ...^] > altchar(0285) # ĝ
```
which requires a hexadecimal to decimal conversion.

I have tested this in practice with trivial changes in 2 lines of code in configUtils.cpp:
``` c
            if (!(c >= '0' && c <= '9' || c >= 'a' && c <= 'f' || c == '+'))
                return false;
            string temp = (c >= 'a' && c <= 'f') ? "" : "NP";

```

With this change I could enter the character of the example above in Notepad. Without, it was impossible.

**Comment by cajhin** [comment]
nice find, I wasn't aware of the +hex numpad combos.
New 85beta supports this.
I did a single test with your +11D example, which worked.

Generally, I'm not really happy with altChar(). It's a crutch, but Windows doesn't offer much else.
Still thinking about sending unicode chars directly to a higher Windows layer, similar to what the virtual keyboard does.
This has two disadvantages I can think of:
1. it gets more complicated. I send characters 'out of order' and have to make sure they are synchronized with the 'real key events'.
2. low level apps (VM, remote, games) might not see these characters
3. it depends on target app support (but so does Alt-NumPad, which is older though and maybe more supported)
